src/main/java/com/rusticisoftware/tincan/Attachment.java

```diff
@@ -79,8 +86,23 @@ public Attachment(JsonNode jsonNode) throws URISyntaxException, MalformedURLExce
         if (! fileUrlNode.isMissingNode()) {
             this.setFileUrl(new URL(fileUrlNode.textValue()));
         }
+
+        JsonNode contentNode = jsonNode.path("content");
+        if (! contentNode.isMissingNode()) {
+            this.setContent(contentNode.binaryValue());
+        }
     }
-    
+
+
+    public void setContent(byte[] content) throws NoSuchAlgorithmException {
+        this.content = Arrays.copyOf(content, content.length);
+        setLength(content.length);
```

Comment:
For code consistency should use `this.setLength(...);` when calling methods internally.